### PR TITLE
systemctl: pretty-print ExtensionImages property

### PR DESCRIPTION
**Pull request by bluca**
Complex type, so without explicit support 'systemctl show' just prints [unprintable]

### CODE REVIEW

File: src/systemctl/systemctl-show.c [hunk 1]
```diff
@@ -1662,6 +1662,62 @@ static int print_property(const char *name, const char *expected_value, sd_bus_m
...
+                                str = strjoin(ignore_enoent ? "-" : "", source);
+                                if (!str)
+                                        return log_oom();
+
+                                r = sd_bus_message_enter_container(m, 'a', "(ss)");
+                                if (r < 0)
+                                        return r;
+
+                                while ((r = sd_bus_message_read(m, "(ss)", &partition, &mount_options)) > 0)
+                                        if (!strextend_with_separator(&str, ":", partition, ":", mount_options))
```
Comment (yuwata) [inline src/systemctl/systemctl-show.c]: This looks strange for me. I guess
```
strextend_with_separator(&str, ":", partition, mount_options)
```
or
```
strextendf_with_separator(&str, ",", "%s:%s", partition, mount_options)
```

File: src/systemctl/systemctl-show.c [hunk 2]
```diff
@@ -1662,6 +1662,62 @@ static int print_property(const char *name, const char *expected_value, sd_bus_m
...
+                                const char *source, *partition, *mount_options;
+                                int ignore_enoent;
+
+                                r = sd_bus_message_enter_container(m, 'r', "sba(ss)");
+                                if (r < 0)
+                                        return r;
+
+                                r = sd_bus_message_read(m, "sb", &source, &ignore_enoent);
+                                if (r <= 0)
+                                        break;
```
Comment (yuwata) [inline src/systemctl/systemctl-show.c]: I think the following is better
```
                                r = sd_bus_message_enter_container(m, 'r', "sba(ss)");
                                if (r < 0)
                                        return r;
                                if (r == 0)
                                        break;

                                r = sd_bus_message_read(m, "sb", &source, &ignore_enoent);
                                if (r < 0)
                                        return r;
```

File: src/systemctl/systemctl-show.c (hunk 1 above)
Comment (bluca) [reply src/systemctl/systemctl-show.c]: was copied from above, will fix both

File: src/systemctl/systemctl-show.c [hunk 3]
```diff
@@ -1635,7 +1637,7 @@ static int print_property(const char *name, const char *expected_value, sd_bus_m
                                         return r;
```
Comment (yuwata) [inline src/systemctl/systemctl-show.c]: `return bus_log_parse_error(r);`

**Pull request merged**